tests/providers/opsgenie/typing/test_opsgenie.py

```diff
@@ -0,0 +1,106 @@
+            {"username": "trinity@opsgenie.com", "type": "user"},
+            {"id": "aee8a0de-c80f-4515-a232-501c0bc9d715", "type": "escalation"},
+            {"name": "Nightwatch Escalation", "type": "escalation"},
+            {"id": "80564037-1984-4f38-b98e-8a1f662df552", "type": "schedule"},
+            {"name": "First Responders Schedule", "type": "schedule"},
+        ],
+        "visible_to": [
+            {"id": "4513b7ea-3b91-438f-b7e4-e3e54af9147c", "type": "team"},
+            {"name": "rocket_team", "type": "team"},
+            {"id": "bb4d9938-c3c2-455d-aaab-727aa701c0d8", "type": "user"},
+            {"username": "trinity@opsgenie.com", "type": "user"},
+        ],
+        "actions": ["Restart", "AnExampleAction"],
+        "tags": ["OverwriteQuietHours", "Critical"],
+        "details": {"key1": "value1", "key2": "value2"},
+        "entity": "An example entity",
+        "source": "Airflow",
+        "priority": "P1",
+        "user": "Jesse",
+        "note": "Write this down",
+    }
+
+    expected_payload_dict = {
+        "message": _config["message"],
+        "alias": _config["alias"],
+        "description": _config["description"],
+        "responders": _config["responders"],
+        "visible_to": _config["visible_to"],
+        "actions": _config["actions"],
+        "tags": _config["tags"],
+        "details": _config["details"],
+        "entity": _config["entity"],
+        "source": _config["source"],
+        "priority": _config["priority"],
+        "user": _config["user"],
+        "note": _config["note"],
+    }
+
+    @mock.patch.object(OpsgenieAlertHook, "get_conn")
+    def test_create_alert_payload_happy_path(self, mock_opsgenie_alert_hook, dag_maker):
+        dag_id = "test_create_alert_payload"
+        with dag_maker(dag_id) as dag:
+            EmptyOperator(task_id="task1")
+
+        notifier = OpsgenieNotifier(payload=self._config)
+        notifier({"dag": dag})
+        mock_opsgenie_alert_hook.return_value.create_alert.assert_called_once_with(self.expected_payload_dict)
+
+    @mock.patch.object(OpsgenieAlertHook, "get_conn")
+    def test_create_alert_payload_unhappy_path(self, mock_opsgenie_alert_hook, dag_maker):
+        dag_id = "test_create_alert_payload"
+        with dag_maker(dag_id) as dag:
+            EmptyOperator(task_id="task1")
+
+        config_no_message = deepcopy(self._config)
+        del config_no_message["message"]
+
+        expected_payload_dict_no_message = deepcopy(self.expected_payload_dict)
+        del expected_payload_dict_no_message["message"]
+
+        with pytest.raises(Exception):  # should raise exception because message is a required key
+            notifier = OpsgenieNotifier(payload=config_no_message)
+            notifier({"dag": dag})
+            mock_opsgenie_alert_hook.return_value.create_alert.assert_called_once_with(
+                expected_payload_dict_no_message
+            )
```

Comment:
Well it would not raise any error because you mock `OpsgenieAlertHook.get_conn`

What we actually want to test here?

Reply:
If `payload` really requires `message` at all times, as it should. Look at my inline comment there
`should raise exception because message is a required key`

Reply:
But this behaviour depend on upstream library which we actually mocks, so this error never happen in tests

Reply:
As result tests failed

Reply:
My point here that we should not tests upstream library exceptions, because it could changed in the future.